cmake/common.cmake

```diff
@@ -93,6 +94,13 @@ if(SAN AND LVI_MITIGATIONS)
   )
 endif()
 
+if(TSAN AND LVI_MITIGATIONS)
+  message(
+    FATAL_ERROR
+      "Building with both TSAN and LVI mitigations is unsafe and deadlocks - choose one"
```

Comment:
It doesn't deadlock, but causes the linker to get stuck in an infinite loop.

Reply:
Thanks for clarifying. This PR is based on https://github.com/microsoft/CCF/pull/5073 and this change is a part of it.
I wanted to use TSAN to check if the changes are OK.